test/test-v8debugapi.js

```diff
@@ -371,6 +371,21 @@ describe('v8debugapi', function() {
 
     });
 
+    it('should work with multiply hit breakpoints', function(done) {
```

Comment:
I don't think this test fails even without your fix above. I think we need a proper regression test for the bug you're fixing. I think you need to add a try-catch around the body of handleDebugEvents, and (somehow) to ensure that it actually never catches anything.